### PR TITLE
Electron 2

### PR DESCRIPTION
**Pull request by Stanzilla**
Just to get a sense what is needed to get it to run and get some CI builds for testing.

* We need a CSP now (https://electronjs.org/docs/tutorial/security#6-define-a-content-security-policy)
* electron-devtools-installer is not yet updated (https://github.com/MarshallOfSound/electron-devtools-installer/issues/73)
* .setZoomLevelLimits got replaced by setVisualZoomLevelLimits (https://github.com/electron/electron/pull/11974/)

Electron 2.0.0 Breaking Changes Overview: https://github.com/electron/electron/blob/v2.0.0/docs/tutorial/planned-breaking-changes.md

### CODE REVIEW

**Comment by MarshallOfSound** [comment]
@Stanzilla `electron-devtools-installer` now supports >=  2.0 👍

**Comment by Stanzilla** [comment]
@MarshallOfSound thanks! pushed that.

Current state in general: the menu does not work.

**Comment by Stanzilla** [comment]
This PR now (temporarily) includes xterm 3.4.0 to test interaction with electron v2 and in general.

**Comment by Stanzilla** [comment]
@chabou ping

**Comment by chabou** [comment]
Will be merged in 2.2.0 except if it brings some awesome perf improvement or bug fix

File: .gitignore [hunk 1]
```diff
@@ -14,3 +14,4 @@ yarn-error.log
 .hyper.js
 .hyper_plugins
 
+.vscode/settings.json
```
Comment (albinekb) [inline .gitignore]: Is this needed?

File: .travis.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ language: node_js
 matrix:
   include:
     - os: linux
-      node_js: 8
+      node_js: 8.9.3
```
Comment (albinekb) [inline .travis.yml]: Is this what electron uses  in 2.0.8?

Comment (Stanzilla) [reply .travis.yml]: Hard to tell, their version numbers in files are all over the place but that is the one they used at the start of 2.0.x

File: .gitignore (hunk 1 above)
Comment (Stanzilla) [reply .gitignore]: "needed", contributors will have this file if they work with VSCode so ignoring it makes sure it is not commited by accident

**Comment by Stanzilla** [comment]
@chabou what do you want to do?

File: .travis.yml (hunk 2 above)
Comment (Stanzilla) [reply .travis.yml]: there is a table now https://github.com/electron/node#branching-strategy

**Comment by chabou** [comment]
Electron v3 is out.

**Comment by Stanzilla** [comment]
Even more reason to merge this now?

File: .gitignore (hunk 1 above)
Comment (sindresorhus) [reply .gitignore]: This should be in your own global gitignore. Only project related ignores should be here.

https://gist.github.com/subfuzion/db7f57fff2fb6998a16c

**Pull request closed without merging**